sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-resource-server/README.md

```diff
@@ -0,0 +1,117 @@
+2. Find the `appRoles` configuration item, and add the following configuration, then click the **Save** button.
+```json
+  {
+    "allowedMemberTypes": [
+      "Application"
+    ],
+    "description": "Task.read",
+    "displayName": "Task.read",
+    "id": "d2bec026-b75f-418d-9493-8462f54f25d9",
+    "isEnabled": true, 
+    "value": "Test.read"
+  },
+  {
+    "allowedMemberTypes": [
+      "Application"
+    ],
+    "description": "Task.wirte",
+    "displayName": "Task.wirte",
+    "id": "1ab4eeda-d07e-4bce-8f77-b0a84c97c34f",
+    "isEnabled": true,
+    "value": "Test.wirte"
+  }
+```
+
+![Configuration Application Roles](docs/image-configuration-application-roles.png "Configuration Application Roles")
+
+3. Find the application permissions need to use.
+
+![Selected Application](docs/image-selected-application.png "Selected Application")
+![Add Application Roles](docs/image-add-application-roles.png "Add Application Roles")
+
+4. Consent Application permissions.
+
+![Consent Application permissions](docs/image-consent-application-permissions.png "Consent Application permissions")
+
+5. In the end, configuration is as follows.
+   
+![Final Configuration](docs/image-final-configuration.png "Final Configuration")
+
+## Examples
+### Configure the sample
+#### application.yml
+
+```yaml
+# In v2.0 tokens, this is always the client ID of the API, while in v1.0 tokens it can be the resource URI used in the request.
+# If we configure azure.activedirectory.b2c.app-id-uri will be to check the audience.
+# If you are using v1.0 tokens, configure app-id-uri to properly complete the audience validation.
+
+azure:
+  activedirectory:
+    b2c:
+      client-id: ${your-client-id}
+      tenant-id: ${your-tenant-id}
+      app-id-uri: ${your-app-id-uri}         # If you are using v1.0 tokens, configure app-id-uri to properly complete the audience validation. 
+```
+
+### Run with Maven
+```
+cd azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-resource-server
+mvn spring-boot:run
+```
+
+### Access the Web API
+We could use Postman to simulate a Web APP to send a request to a Web API.
+**NOTE**: The `aud` in access token should be client id of the current application.
+
+```http request
+GET /write HTTP/1.1
+Authorization: Bearer eyJ0eXAiO ... 0X2tnSQLEANnSPHY0gKcgw
+```
+```http request
+GET /read HTTP/1.1
+Authorization: Bearer eyJ0eXAiO ... 0X2tnSQLEANnSPHY0gKcgw
+```
+
+## Troubleshooting
+- `Missing 'tenant-id' in application.yml`
+    
+    ```
+    Method springSecurityFilterChain in org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration required a bean of type 'org.springframework.security.oauth2.jwt.JwtDecoder' that could not be found.
+        The following candidates were found but could not be injected:
+        - Bean method 'jwtDecoder' in 'AADResourceServerConfiguration' not loaded because @ConditionalOnResource did not find resource 'classpath:aad.enable.config'
```

Comment:
I suppose this is for the AAD starter?
> Bean method 'jwtDecoder' in '**AADResourceServerConfiguration**' not loaded because @ConditionalOnResource did not find resource 'classpath:aad.enable.config'